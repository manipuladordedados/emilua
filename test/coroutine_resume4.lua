-- That's similar to LuaJIT's CORUN error. The runtime can further explore this
-- restriction (user can no longer do shenanigans to wake up not-ready fibers)
-- to make implementation way easier.
local println = require('println')

shared = coroutine.create(function() this_fiber.yield() end)

spawn(function()
    -- Conceptually, coroutine `shared` is still running when this fiber
    -- suspends. Therefore, it is only natural to make the implementation follow
    -- the concept.
    coroutine.resume(shared)
end)

println(coroutine.status(shared))
this_fiber.yield()
println(coroutine.status(shared))
coroutine.resume(shared)
