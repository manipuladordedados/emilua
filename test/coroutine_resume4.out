suspended
normal
Main fiber from VM 0x1 panicked: 'Operation not permitted'
stack traceback:
	[string "?"]: in function 'resume'
	input.lua:16: in main chunk
	[C]: in function ''
	[string "?"]: in function <[string "?"]:0>
